crates/krator/src/store.rs

```diff
@@ -0,0 +1,82 @@
+//! Defines Store type for caching Kubernetes objects locally.
+
+use crate::object::ObjectKey;
+use std::any::Any;
+use std::collections::HashMap;
+use tokio::sync::RwLock;
+
+#[derive(PartialEq, Eq, Hash)]
+/// Redefinition of Resource to implement Eq and Hash.
+pub(crate) struct ResourceKey {
+    api_version: String,
+    group: String,
+    kind: String,
+    version: String,
+    namespace: Option<String>,
+}
+
+impl From<kube::Resource> for ResourceKey {
+    fn from(resource: kube::Resource) -> Self {
+        ResourceKey {
+            api_version: resource.api_version,
+            group: resource.group,
+            kind: resource.kind,
+            version: resource.version,
+            namespace: resource.namespace,
+        }
+    }
+}
+
+type ResourceMap = HashMap<ResourceKey, HashMap<ObjectKey, Box<dyn Any>>>;
+
+/// Stores or caches arbitrary Kubernetes objects.
+pub struct Store {
+    objects: RwLock<ResourceMap>,
+}
+
+impl Store {
+    /// Insert a new object.
+    pub async fn insert<R: 'static + k8s_openapi::Resource>(
```

Comment:
You could avoid `async` on both functions and a `RwLock` if you make this an `&mut self`

Reply:
I'm expecting `Store` will be stored in an `Arc`, so I would prefer to only require `&self` and hide the `RwLock` access inside `insert` and `get`.

Reply:
Ok, sounds good to me